include/deal.II/base/parameter_handler.h

```diff
@@ -1997,6 +1989,60 @@ class ParameterHandler : public Subscriptor
   DeclException1 (ExcEntryUndeclared,
                   std::string,
                   << "You can't ask for entry <" << arg1 << "> you have not yet declared.");
+
+  /**
+   * Exception
+   */
+  DeclException1 (ExcUnbalancedSubsections,
+                  std::string,
+                  << "There are unequal numbers of 'subsection' and 'end' "
+                  "statements in the parameter file <" << arg1 << ">.");
+
+  /**
+   * Exception
+   */
+  DeclException3 (ExcNoSubsection,
+                  int, std::string, std::string,
+                  << "Line <" << arg1 << "> of file <" << arg2 << ": There is "
+                  "no such subsection to be entered: " << arg3);
+
+  /**
+   * Exception
+   */
+  DeclException3 (ExcCannotParseLine,
+                  int, std::string, std::string, << "Line <" << arg1 <<
+                  "> of file <" << arg2 << ">: " << arg3);
```

Comment:
Oh, looking at it in the actual code, I realize that the third argument is the error text. I think that's unusual. I'm ok with that, but it may be useful saying in the comment before the exception that the third argument is the actual error message.